crates/ruff_cli/src/commands/format.rs

```diff
@@ -0,0 +1,104 @@
+use colored::Colorize;
+use rayon::iter::{IntoParallelIterator, ParallelIterator};
+use ruff::resolver::python_files_in_path;
+use ruff::settings::types::{FilePattern, FilePatternSet};
+use ruff_formatter::LineWidth;
+use ruff_python_formatter::{format_module, FormatModuleError, PyFormatOptions};
+use std::io;
+use std::path::{Path, PathBuf};
+use thiserror::Error;
+use tracing::span;
+use tracing::Level;
+
+/// The inner errors are all flat, i.e. none of them has a source
+#[derive(Error, Debug)]
+enum FormatterIterationError {
+    #[error("Failed to traverse the inputs paths: {0}")]
+    Ignore(#[from] ignore::Error),
+    #[error("Failed to read {0}: {1}")]
+    Read(PathBuf, io::Error),
+    #[error("Failed to write {0}: {1}")]
+    Write(PathBuf, io::Error),
+    #[error("Failed to format {0}: {1}")]
+    FormatModule(PathBuf, FormatModuleError),
+}
+
+pub(crate) fn format(cli: &Arguments, overrides: &Overrides) -> anyhow::Result<ExitStatus> {
+    let mut pyproject_config = resolve(
+        cli.isolated,
+        cli.config.as_deref(),
+        overrides,
+        cli.stdin_filename.as_deref(),
+    )?;
+    // We don't want to format pyproject.toml
+    pyproject_config.settings.lib.include = FilePatternSet::try_from_vec(vec![
+        FilePattern::Builtin("*.py"),
+        FilePattern::Builtin("*.pyi"),
+    ])
+    .unwrap();
+    let (paths, resolver) = python_files_in_path(&cli.files, &pyproject_config, overrides)?;
+    if paths.is_empty() {
+        bail!("no python files in TODO(@konstin) pass them in")
+    }
+
+    let results: Vec<Result<(), FormatterIterationError>> = paths
+        .into_par_iter()
+        .map(|dir_entry| {
+            let dir_entry = dir_entry?;
+            // TODO(@konstin): For some reason it does not filter in the beginning
+            // TODO(@konstin): When integrating with the linter, how do we get the filtering right?
+            let path = dir_entry.path();
+            if dir_entry.file_name() == "pyproject.toml"
+                || path.extension().is_some_and(|ext| ext == "ipynb")
+            {
+                return Ok(());
+            }
+
+            let line_length = resolver.resolve(path, &pyproject_config).line_length;
+            // TODO(@konstin): Unify `LineWidth` and `LineLength`
+            let line_width = LineWidth::try_from(
+                u16::try_from(line_length.get()).expect("Line shouldn't be larger than 2**16"),
+            )
+            .expect("Configured line length is too large for the formatter");
+            let options = PyFormatOptions::from_extension(path).with_line_width(line_width);
+
+            format_path(path, options)
+        })
+        .collect();
+
+    let mut all_success = true;
+    for result in results {
+        if let Err(err) = result {
+            // The inner errors are all flat, i.e. none of them has a source
+            eprintln!("{}", err.to_string().red().bold());
+            all_success = false;
+        }
+    }
```

Comment:
Can we do this more like in `run` where the error handling is inside the `map` above?

Reply:
moved it into the map, i hope that's what you were aiming for